Port exploit to Python 3 and new system:

* chall 10
* chall 13
* chall 16

